OHHTTPStubs/Sources/OHHTTPStubs.m

```diff
@@ -424,7 +424,24 @@ - (void)startLoading
                 // Notify if a redirection occurred
                 if (((responseStub.statusCode > 300) && (responseStub.statusCode < 400)) && redirectLocationURL)
                 {
-                    NSURLRequest* redirectRequest = [NSURLRequest requestWithURL:redirectLocationURL];
+                    NSMutableURLRequest* mReq = [self.request mutableCopy];
+                    switch (responseStub.statusCode)
+                    {
+                        case 301:
+                        case 302:
+                        case 307:
+                        case 308:
+                            //Preserve the original request method and set the new location URL
+                            [mReq setHTTPMethod:self.request.HTTPMethod];
```

Comment:
This isn't needed anymore as we do a `mutableCopy` on line 427 now so the `HTTPMethod` is already the same as the original